### PR TITLE
MAGN-7524 Use OAuth to authenticate submitted workspace

### PR DESCRIPTION
**Pull request by pbidenko**
### Purpose

Youtrack task: [MAGN-7524](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7524) Use OAuth to authenticate submitted workspace.

In this PR I moved out authorization related code from package manager into a separate module.

### Reviewers

@pboyer

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Core/LoginViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+﻿using System.Windows.Input;
```
Comment (pboyer) [inline src/DynamoCoreWpf/ViewModels/Core/LoginViewModel.cs]: Let's remove this.

**Comment by pboyer** [comment]
@pbidenko  See my email for more details.

**Comment by pbidenko** [comment]
Hi @pboyer,

I made those changes. Please, take a look.

**Comment by pboyer** [comment]
@pbidenko This may need to be reworked a bit due to #4670.

**Comment by pbidenko** [comment]
Hi @pboyer 
please, take a look again.

**Comment by aparajit-pratap** [comment]
@pbidenko unfortunately this PR has caused the `Login` button states to regress for Dynamo Studio. The `Login` button is bound to the `LoginState` and the `UserName` properties in `PackageManagerClientViewModel`, which is its `DataContext`.

**Pull request merged**